refactor: add test-rust workflow

fool1280 opened the pull request:
Issue: https://github.com/PRQL/prql/issues/1424
- Move `test-rust` actions to its separate workflow
- Update `test-all.yaml` and `pull-request.yaml` accordingly

fool1280 committed:
update test-rust workflow

fool1280 committed:
fix yaml precommit

fool1280 committed:
Merge branch 'main' into update-test-rust-workflow

fool1280 committed:
test

fool1280 committed:
delete ref for test-rust actions

fool1280 committed:
Merge branch 'update-test-rust-workflow' of github.com:fool1280/prql into update-test-rust-workflow

fool1280 committed:
update

max-sixty commented on .github/workflows/test-rust.yaml:
```diff
@@ -0,0 +1,75 @@
+          - os: windows-latest
+          - os: macos-latest
+          - os: ubuntu-latest
+            target_option: "--target=wasm32-unknown-unknown"
```
For PRs, we want these to run really quickly, so we only run on ubuntu, which catches all but the rarest errors (maybe 1/1000 errors are only on windows). Then on `test-all` we want to run all the OSs.

So I think it might be worth going back to defining this list of OSs in `test-all` and passing the values to `test-rust`.
`pull-request` already defines the two options it wants for `target_option`, so that should work.

Does that make sense? Lmk if it's confusing...

fool1280 replied:
I understand it now, let me fix it

fool1280 committed:
Merge branch 'main' into update-test-rust-workflow

fool1280 committed:
update input for test-rust

fool1280 committed:
Merge branch 'update-test-rust-workflow' of github.com:fool1280/prql into update-test-rust-workflow

fool1280 committed:
update pull-request

fool1280 committed:
Merge branch 'main' into update-test-rust-workflow

fool1280 committed:
pull-request try1

fool1280 committed:
:wq
Merge branch 'update-test-rust-workflow' of github.com:fool1280/prql into update-test-rust-workflow

fool1280 committed:
pr try2

fool1280 committed:
try3

fool1280 committed:
try4

fool1280 committed:
Merge branch 'main' into update-test-rust-workflow

fool1280 committed:
try5

fool1280 committed:
try6

fool1280 committed:
finalize

max-sixty commented on .github/workflows/test-all.yaml:
```diff
@@ -22,21 +22,14 @@ concurrency:
-        with:
-          target_option: ${{ matrix.target_option }}
+        os: [ubuntu-latest, windows-latest, macos-latest, ubuntu-latest]
+        target_option: [--target=wasm32-unknown-unknown]
```
```suggestion
        target_option: ["", --target=wasm32-unknown-unknown]
```

max-sixty committed:
Update .github/workflows/test-all.yaml

max-sixty commented on .github/dependabot.yml:
```diff
@@ -45,13 +45,6 @@ updates:
     schedule:
       interval: "daily"
 
-  - directory: ".github/actions/test-rust"
```
Can we also remove the file at `".github/actions/test-rust"`?

max-sixty commented on .github/workflows/test-rust.yaml:
```diff
@@ -0,0 +1,76 @@
+        type: string
+      target_option:
+        required: false
+        default: --target=wasm32-unknown-unknown
```
```suggestion
        default: ""
```

(and to the extent there was a description of this field in the original, would be great if we can copy it here as a comment)

fool1280 replied:
Can I ask more context about which comments? I'm seeing the description of the target-option or this one: 

`Without inputs.target, it seems to select linux x64 in the wasm32 tests, though in the docs it says it uses the target as the key?`

fool1280 committed:
Update .github/workflows/test-rust.yaml

Co-authored-by: Maximilian Roos <5635139+max-sixty@users.noreply.github.com>

fool1280 committed:
Merge branch 'main' into update-test-rust-workflow

fool1280 committed:
update test-js, test-java

max-sixty committed:
try getting around deadlock again

max-sixty committed:
try 'manual' default value for os in test-rust

max-sixty committed:
.

max-sixty committed:
.

max-sixty committed:
shorter task name

max-sixty committed:
remove duplicate ubuntu

max-sixty committed:
comment

max-sixty committed:
.
